lib/web_ui/test/engine/semantics/accessibility_test.dart

```diff
@@ -43,76 +55,89 @@ void testMain() {
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'polite message');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
+    test('aria-live is polite when assertiveness is null', () async {
+      sendAnnouncementMessage(message: 'polite message');
+      expectMessages(polite: 'polite message');
+
+      await Future<void>.delayed(liveMessageDuration);
+      expectNoMessages();
     });
 
-    test('aria-live is polite when assertiveness is set to 0', () {
-      resetAccessibilityAnnouncements();
-      const Map<dynamic, dynamic> testInput = <dynamic, dynamic>{'data': <dynamic, dynamic>{'message': 'polite message', 'assertiveness': 0}};
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'polite message');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
+    test('aria-live is polite when assertiveness is set to 0', () async {
+      sendAnnouncementMessage(message: 'polite message', assertiveness: 0);
+      expectMessages(polite: 'polite message');
+
+      await Future<void>.delayed(liveMessageDuration);
+      expectNoMessages();
     });
 
-    test('The same message announced twice is altered to convince the screen reader to read it again.', () {
-      resetAccessibilityAnnouncements();
-      const Map<dynamic, dynamic> testInput = <dynamic, dynamic>{'data': <dynamic, dynamic>{'message': 'Hello'}};
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'Hello');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
-
-      // The DOM value gains a "." to make the message look updated.
-      const Map<dynamic, dynamic> testInput2 = <dynamic, dynamic>{'data': <dynamic, dynamic>{'message': 'Hello'}};
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput2));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'Hello.');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
-
-      // Now the "." is removed because the message without it will also look updated.
-      const Map<dynamic, dynamic> testInput3 = <dynamic, dynamic>{'data': <dynamic, dynamic>{'message': 'Hello'}};
-      accessibilityAnnouncements.handleMessage(codec, codec.encodeMessage(testInput3));
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.polite).text, 'Hello');
-      expect(accessibilityAnnouncements.ariaLiveElementFor(Assertiveness.assertive).text, '');
+    test('Rapid-fire messages are each announced.', () async {
```

Comment:
Does this behavior change if mixed assertive/polite rapid-fire messages are sent, rather than just polite?

Reply:
Nope. Nothing fundamentally changes. They go into different live divs, each of which is handled independently. These tests ensure that multiple messages of the same type don't clobber each other in any way.